xbmc/addons/GUIViewStateAddonBrowser.cpp

```diff
@@ -40,6 +41,15 @@ CGUIViewStateAddonBrowser::CGUIViewStateAddonBrowser(const CFileItemList& items)
   else
   {
     AddSortMethod(SortByLabel, SortAttributeIgnoreFolders, 551, LABEL_MASKS("%L", "%I", "%L", ""));  // Filename, Size | Foldername, empty
+
+    if (URIUtils::PathStarts(items.GetPath(), "addons://sources/"))
```

Comment:
Can you get back to me on slack or mail? Hard to explain here. Basically if you enter a path like the above you can filter by name or last used.
The first time you toggle through is fine, when you get back to title sorting at the end of the list, the sorting will be alphabetically again. And when you switch to last used again, everything stays alphabetically sorted.